flaml/default/portfolio.py

```diff
@@ -0,0 +1,226 @@
+import pandas as pd
+import numpy as np
+import argparse
+from pathlib import Path
+import json
+import sys
+from sklearn.preprocessing import RobustScaler
+from flaml.default import greedy
+from flaml.default.regret import load_result, build_regret
+
+regret_bound = 0.01
+
+
+def config_predictor_tuple(tasks, configs, meta_features, regret_matrix):
+    """Config predictor represented in tuple."""
+    # pre-processing
+    scaler = RobustScaler()
+    meta_features = meta_features.loc[tasks]  # this makes a copy
+    meta_features.loc[:, :] = scaler.fit_transform(meta_features)
+
+    proc = {
+        "center": scaler.center_.tolist(),
+        "scale": scaler.scale_.tolist(),
+    }
+
+    # best model for each dataset in training
+    # choices = regret_matrix[tasks].loc[configs].reset_index(drop=True).idxmin()
+
+    # break ties using the order in configs
+    regret = (
+        regret_matrix[tasks]
+        .loc[configs]
+        .reset_index(drop=True)
+        .apply(lambda row: row.apply(lambda x: (x, row.name)), axis=1)
+    )
+    print(regret)
+    preferences = np.argsort(regret, axis=0)
+    print(preferences)
+    return (meta_features, preferences, proc)
+
+
+def build_portfolio(meta_features, regret, strat):
+    """Build a portfolio using our algorithm.
+
+    Args:
+        meta_features: metafeatures matrix
+        regret: regret matrix
+        strat: one of {greedy-feedback, greedy}
+    """
+    if strat == "greedy-feedback":
+        portfolio = greedy.smart(regret, meta_features, regret_bound)
+    elif strat == "greedy":
+        portfolio = greedy.smart(regret, None, regret_bound)
```

Comment:
What if strat is neither 'greedy' nor 'greedy-feedback'? Add a default setting for strat?